post-payments.php

```diff
@@ -0,0 +1,197 @@
+					<input type="submit" value="<?php esc_html_e( 'Generate' ); ?>" class="button submit" />
+				</form>
+			</p>
+		</div><?php
+
+		if ( ! empty( $_GET['from_date'] ) && ! empty( $_GET['to_date'] ) ) {
+			$authors = $this->get_report_data( sanitize_text_field( $_GET['from_date'] ), sanitize_text_field( $_GET['to_date'] ) );
+
+
+			if ( empty( $authors ) ) {
+				?><p><b><?php esc_html_e( 'No posts found with cost data in this date range.' ); ?></b></p><?php
+			} else {
+				?>
+
+				<?php foreach ( $authors as $name => $data ): ?>
+					<p class="post-payments-author-name"><b><?php echo esc_html( $name ); ?></b></p>
+					<ul class="post-payments-list"><?php foreach ( $data['posts'] as $post_id ): ?>
+						<li><?php echo esc_html( $this->format_currency( get_post_meta( $post_id, $this->meta_key, true ) ) ); ?>: <i><a href="<?php echo esc_url( get_permalink( $post_id ) ); ?>"><?php echo esc_html( get_the_title( $post_id ) ); ?></a></i></li>
+					<?php endforeach; ?></ul>
+					<p class="post-payments-author-payment"><?php esc_html_e( 'Total payment:' ); ?> <b><?php echo esc_html( $this->format_currency( $data['total'] ) ); ?></b></p>
+				<?php endforeach; ?>
+
+				<?php
+			}
+		}
+	}
+
+	public function get_post_types() {
+		$settings = get_option( $this->settings_option );
+		if ( empty( $settings['post_types'] ) ) {
+			return $this->default_post_types;
+		} else {
+			return $settings['post_types'];
+		}
+	}
+
+	public function get_report_data( $from_date, $to_date ) {
+		$posts = get_posts( array(
+			'posts_per_page' => -1,
+			'post_status' => 'publish',
+			'post_type' => $this->get_post_types(),
+			'meta_query' => array(
+				array(
+					'key' => $this->meta_key,
+					'compare' => 'EXISTS',
+				),
+			),
+			'date_query' => array(
+				array(
+					'after' => $from_date,
+					'before' => $to_date,
+				),
+			),
+		) );
+
+		$authors = array();
+
+		foreach ( $posts as $post ) {
+			$post_authors = get_coauthors( $post->ID );
+			foreach ( $post_authors as $post_author ) {
+				if ( empty( $authors[$post_author->display_name] ) ) {
+					$authors[$post_author->display_name] = $this->add_author( $post_author, $post->ID );
+				} else {
+					$authors[$post_author->display_name] = $this->add_author( $post_author, $post->ID, $authors[$post_author->display_name] );
+				}
+			}
+		}
+
+		return $authors;
+	}
+
+	public function add_author( $post_author, $post_id, $data = null ) {
+		if ( ! $data ) {
+			$data = array(
+				'posts' => array(),
+				'payments' => array(),
+			);
+		}
+
+		$data['posts'][] = $post_id;
+
+		$payment = get_post_meta( $post_id, $this->meta_key, true );
+		if ( is_numeric( $payment ) ) {
+			$data['payments'][] = floatval( $payment );
+		}
+
+		$data['total'] = array_sum( $data['payments'] );
+
+		return $data;
+
+	}
+
+	public function format_currency( $number ) {
+		return $this->currency_symbol . number_format( $number, 2 );
+	}
+
+	
+}
+
+new Post_Payments();
```

Comment:
It might make sense to run this during init or something like that so that the get_options call isn't running quite as early